### DISCUSSION TITLE
Import edge from csv file?

### DISCUSSION THREAD

**Discussion by Alex-Li96** [networkit/networkit#1082] [Q&A] [answered]
I am trying to compute PageRank of a large graph and the edge data is stored in CSV format. The only way I managed to work was first creating a graph, and then importing each pair of edge into the graph using for loop. 
```
G = nk.Graph(node.shape[0], directed = True, weighted = False)
for i in edgeint.index:
  G.addEdge(edge.from_id[i], edge.to_id[i])
pr = nk.centrality.PageRank(G, damp = 0.75, tol = 1e-19)  
```
Importing data took 10+ hrs since I got 8 billions + edges. Is there anyways to import the csv edge file directly and make it works?
Or is there anyway to make the process faster?

**Comment by fabratu** [comment, marked as answer]
Hi, thanks for reaching out. 

My first try would be to use the EdgelistReader[1] from NetworKit. The separator is likely `,` and you have to remove - if present - any additional columns beside `u`,`v` and `w` (both endpoints and the weight).

If that works, you can also convert the graph to NetworKitBinaryFormat[2], which saves disk space and is much faster for future I/O related tasks.

Btw - I am converting this to a discussion, since it is more fitting in Q&A.

[1] https://networkit.github.io/dev-docs/python_api/graphio.html?highlight=edgelistreader#networkit.graphio.EdgeListReader
[2] https://networkit.github.io/dev-docs/python_api/graphio.html?highlight=networkitbinary#networkit.graphio.NetworkitBinaryWriter

**Reply by Alex-Li96** [reply]
Thanks for the reply, fabratu.

Based on your reply, I managed to import the csv file to Networkit and I was trying to get the PageRank score base on that. But I am not able match the Ids of the PageRank result to my original data. I noticed that networkit changed the original Ids to new ones, and I failed to obtains the whole list edges recreated by networkit. Is there a way to get the whole list of edges in the form of pd dataframe so I can merge the dataset to get the PR result using my original Ids? Or is there any other ways to get the node list with both the old and the new Ids?

Many thanks.
